src/routes/StudentRequests/StudentRequests.tsx

```diff
@@ -0,0 +1,285 @@
+  }, [filterValue]);
+
+  // Get information from db on requests
+  useEffect(() => {
+    if (user?.role !== 'teacher') {
+      navigate('/');
+    }
+    dispatch(setLoading());
+    setIsLoading(true);
+    const getStudentRequestList = async () => {
+      const response = await getStudentRequests(
+        user?.authToken as string,
+        user?.id as string
+      );
+      console.log(response.data);
+      if (response.status === 'success') {
+        setData(() => response.data as StudentRequest[]);
+      } else {
+        dispatch(updateToast(TOAST_GENERAL_ERRORS.SYSTEM));
+      }
+      dispatch(removeLoading());
+      setIsLoading(false);
+    };
+    getStudentRequestList();
+  }, [updateFetch]);
+
+  // Get requests sorted by buttons and filtered by values
+  const sortedData = sortStudents(
+    data,
+    sortRule.element,
+    sortRule.value
+  ).filter((request) => {
+    if (filterValue === ALL_GROUPS || isListOpen) return request;
+    const row = `${request.subject_id}.${request.class_id}`;
+    return row === filterValue;
+  });
+
+  const isAllSelected =
+    selectedRows.length === sortedData.length && sortedData.length > 0;
+
+  // To display on dropdown menu all unique groups from students
+  const allGroups = [
+    ALL_GROUPS,
+    ...data
+      .map((request) => `${request.subject_id}.${request.class_id}`)
+      .filter((value, index, array) => array.indexOf(value) === index),
+  ];
+
+  // Closes autocomplete list on tab press
+  const skipAutocomplete = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Tab') setIsListOpen(false);
+  };
+
+  //  Handles which rows are selected
+  function handleSelect(row: string) {
+    const index = selectedRows.indexOf(row);
+    if (index === -1) {
+      setSelectedRows([...selectedRows, row]);
+    } else {
+      const newRows = [...selectedRows];
+      newRows.splice(index, 1);
+      setSelectedRows(newRows);
+    }
+  }
+
+  function handleSelectAll() {
+    if (isAllSelected) {
+      setSelectedRows([]);
+    } else {
+      const allSelectedRows = sortedData.map(
+        (request) =>
+          `${request.subject_id}<${request.class_id}<${request.student_id}`
+      );
+      setSelectedRows(allSelectedRows);
+    }
+  }
+
+  // When user chooses to filter per group
+  function selectGroup(group: string) {
+    if (autoComplete.current) autoComplete.current.value = group;
+    setFilterValue(group);
+    setIsListOpen(false);
+    setSelectedRows([]);
+  }
+
+  const onFilterChange = (_id: string, value: string) => {
+    setFilterValue(value);
+  };
+
+  /* eslint-disable @typescript-eslint/naming-convention */
+  function extractRowInfo(row: string) {
+    const values = row.split('<');
+    const subject_id = values[0];
+    const class_id = values[1];
+    const student_id = values[2];
+    return { subject_id, class_id, student_id };
+  }
+
+  // Respond request and send it to db
+  async function handleAction(
```

Comment:
For callbacks, should we use Functions or arrow functions?. Let's try to use just one.